src/awkward/_nplikes/cupy.py

```diff
@@ -165,3 +165,9 @@ def is_c_contiguous(self, x: ArrayLike) -> bool:
             return True
         else:
             return x.flags["C_CONTIGUOUS"]
+
+    def broadcast_arrays(self, *arrays):
+        return [
+            self._module.array(x) if not isinstance(x, self._module.ndarray) else x
+            for x in super().broadcast_arrays(*arrays)
+        ]
```

Comment:
I think this should be handled at the call site - we want arguments to nplikes to be arrays already, for simplicity.